widget/view.controller.js

```diff
@@ -0,0 +1,146 @@
+
+  function threatIntelManagementConfiguration100Ctrl($scope, threatIntelManagementConfigurationService, WizardHandler, $controller, $state, connectorService, currentPermissionsService, CommonUtils, $window, API, _, $filter, $q, dataIngestionService, PagedCollection, $resource, FIXED_MODULE, Entity, playbookService, translationService, toaster, appModulesService, widgetBasePath) {
+    $controller('BaseConnectorCtrl', {
+      $scope: $scope
+    });
+    $scope.backToStartPage = backToStartPage;
+    $scope.moveToConfigureConnector = moveToConfigureConnector;
+    $scope.backToinstallConnector = backToinstallConnector;
+    $scope.moveToSelectConnector = moveToSelectConnector;
+    $scope.installConnector = installConnector;
+    $scope.toggleSelectFeedsSettings = toggleSelectFeedsSettings;
+    $scope.allConnectorsInstalled = false;
+    $scope.feedConnectors = [];
+    $scope.installedConnectors = [];
+    $scope.searchQuery = '';
+    $scope.widgetBasePath = widgetBasePath;
+    $scope.widgetCSS = widgetBasePath + 'widgetAssets/css/wizard-style.css';
+
+    init();
+
+    function init() {
+      $scope.isFortiGuardConectorInsttalled = true;
+      appModulesService.load(true).then(function (modules) {
+        modules = $filter('playbookModules')(modules);
+        $scope.modules = currentPermissionsService.availablePermissions(modules, 'create');
+      });
+    }
+
+    function toggleSelectFeedsSettings(event) {
+      event.stopPropagation();
+    }
+
+    function installConnector() {
+      WizardHandler.wizard('timSolutionpackWizard').next();
+      $scope.getSelectedConnectorsCount = function () {
+        var feedConnectors = Array.isArray($scope.feedConnectors) ? $scope.feedConnectors : [];
+        var installedConnectors = Array.isArray($scope.installedConnectors) ? $scope.installedConnectors : [];
+
+        return feedConnectors.filter(function (connector) {
+          return connector.selectConnector;
+        }).length + installedConnectors.length;
+      };
+      const selectedConnectors = $scope.feedConnectors.filter(pack => pack.selectConnector === true && pack.installed === false);
+
+      function installSequentially(index) {
+        if (index >= selectedConnectors.length) {
+          $scope.allConnectorsInstalled = true;
+          return;
+        }
+        const connector = selectedConnectors[index];
+        threatIntelManagementConfigurationService.installConnector(connector)
+          .then(resp => {
+            const importJobId = resp.data.importJob;
+            threatIntelManagementConfigurationService.getInstallationProgress(importJobId, connector)
+              .then(() => {
+                connector.installed = true;
+                installSequentially(index + 1);
+              })
+              .catch(error => {
+                console.log(error);
```

Comment:
You will also need to store this failure logs and it will help you in showing it in a summary page

Reply:
its failed then we will never reached to summary page. Hence as UI standard it will populated on console